PlayCover/Model/AppSettings.swift

```diff
@@ -7,287 +7,86 @@ import Foundation
-            dictionary = dict
-        }
-    }
-
-    private static let aspectRatio = "pc.aspectRatio"
-    var aspectRatio: Int {
-        get {
-            return dictionary[AppSettings.aspectRatio] as? Int ?? 1
-        }
-        set {
-            var dict = dictionary
-            dict[AppSettings.aspectRatio] = newValue
-            dictionary = dict
-        }
-    }
-
-    private static let notch = "pc.hasNotch"
-    var notch: Bool {
-        get {
-            return NSScreen.hasNotch()
-        }
-        set {
-            var dict = dictionary
-            dict[AppSettings.notch] = newValue
-            dictionary = dict
-        }
-    }
-
-    private static let bypass = "pc.bypass"
-    var bypass: Bool {
-        get {
-            return dictionary[AppSettings.bypass] as? Bool ?? false
-        }
-        set {
-            var dict = dictionary
-            dict[AppSettings.bypass] = newValue
-            dictionary = dict
-        }
-    }
+struct AppSettingsData: Codable {
+    var keymapping: Bool = true
+    var mouseMapping: Bool = true
+    var sensitivity: Float = 50
+
+    var disableTimeout: Bool = false
+    var iosDeviceModel: String = "iPad8,6"
+    var refreshRate: Int = 60
+    var windowWidth: Int = 1920
+    var windowHeight: Int = 1080
+    var resolution: Int = 2
+    var aspectRatio: Int = 1
+    var notch: Bool = NSScreen.hasNotch()
+    var bypass: Bool = false
+}
 
-    private var allPrefs: [String: Any] {
-        get {
-            do {
-                if let all = try [String: Any].read(AppSettings.settingsUrl) {
-                    if !all.isEmpty {
-                        return all
-                    }
-                }
-            } catch {
-                Log.shared.error(error)
-            }
-            return [:]
-        }
-        set {
+class AppSettings {
+    static var appSettingsDir: URL {
+        let settingsFolder =
+            PlayTools.playCoverContainer.appendingPathComponent("App Settings")
+        if !fileMgr.fileExists(atPath: settingsFolder.path) {
             do {
-                try newValue.store(AppSettings.settingsUrl)
+                try fileMgr.createDirectory(at: settingsFolder, withIntermediateDirectories: true, attributes: [:])
             } catch {
                 Log.shared.error(error)
             }
         }
+        return settingsFolder
     }
 
-    private var dictionary: [String: Any] {
-        get {
-            if let prefs = allPrefs[info.bundleIdentifier] as? [String: Any] {
-                return prefs
-            } else {
-                return [AppSettings.keymapping: info.isGame]
-            }
-        }
-        set {
-            var prefs = allPrefs
-            prefs[info.bundleIdentifier] = newValue
-            allPrefs = prefs
+    let info: AppInfo
+    let settingsUrl: URL
+    var container: AppContainer?
+    var settings: AppSettingsData {
+        didSet {
+            _ = encode()
```

Comment:
I think using `@discardableResult` to suppress warning is better. (https://github.com/PlayCover/PlayCover/pull/178)